### PR TITLE
[MB-1635] Update Android Sample App

### PR DESCRIPTION
**Pull request by madelgi**
Updates the Android Xamarin sample app to match the standard Android sample.

### CODE REVIEW

File: samples/android/HomeFragment.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+			channelId = (TextView)view.FindViewById(Resource.Id.channel_id);
+			shareButton = (Button)view.FindViewById(Resource.Id.share_button);
+			copyButton = (Button)view.FindViewById(Resource.Id.copy_button);
+
+			copyButton.Click += (sender, e) =>
+			{
+				if (!string.IsNullOrEmpty(channelId.Text))
+				{
+					ActionRunRequest.CreateRequest(ClipboardAction.DefaultRegistryName)
+									.SetValue(UAirship.Shared().PushManager.ChannelId)
```
Comment (rlepinski) [inline samples/android/HomeFragment.cs]: fromatting here seems messed up

File: samples/android/LocationFragment.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+			View view = inflater.Inflate(Resource.Layout.fragment_location, container, false);
+
+			priorityGroup = (RadioGroup)view.FindViewById(Resource.Id.location_priority);
+			progress = view.FindViewById(Resource.Id.request_progress);
+			progress.Visibility = ViewStates.Invisible;
+
+			Button button = (Button)view.FindViewById(Resource.Id.request_button);
+			button.Click += (sender, e) =>
+			{
+				// RequestLocation();
```
Comment (rlepinski) [inline samples/android/LocationFragment.cs]: unfinshed code?

File: samples/android/Preference/SetAliasPreference.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+using Android.Preferences;
+using Android.Runtime;
+using Android.Util;
+using Android.Views;
+using Android.Widget;
+
+using UrbanAirship;
+
+// TODO alias stuff, getter setter blah;
+namespace Sample.Preference
```
Comment (rlepinski) [inline samples/android/Preference/SetAliasPreference.cs]: We can remove alias

**Comment by rlepinski** [comment]
👍  This is amazing

File: samples/android/HomeFragment.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+			};
+
+			return view;
+		}
+
+		public override void OnResume()
+		{
+			base.OnResume();
+
+			// Register a local broadcast manager to list for ACTION_UPDATE_CHANNEL
```
Comment (hcrowell) [inline samples/android/HomeFragment.cs]: ... to listen* for ...

File: samples/android/MainActivity.cs [hunk 5]
```diff
@@ -3,98 +3,290 @@
...
-		private TextView channelID;
-		private BroadcastReceiver channelIdUpdateReceiver;
+		private const string Tag = "MainActivity";
 
-		protected override void OnCreate (Bundle bundle)
-		{
-			base.OnCreate (bundle);
+		private const string NavId = "NAV_ID";
+		private const string TitleKey = "TITLE";
+		private const string LastMessageCentDate = "LAST_MC_SENT_DATE";
```
Comment (hcrowell) [inline samples/android/MainActivity.cs]: LastMessageSentDate* ?

File: samples/android/MainActivity.cs [hunk 6]
```diff
@@ -3,98 +3,290 @@
...
+			base.OnNewIntent(intent);
+			Intent = intent;
+		}
+
+		protected override void OnSaveInstanceState(Bundle outState)
+		{
+			base.OnSaveInstanceState(outState);
+			outState.PutInt(NavId, currentNavPosition);
+			outState.PutString(TitleKey, Title);
+			outState.PutLong(LastMessageCentDate, messageCenterLastSentDate);
```
Comment (hcrowell) [inline samples/android/MainActivity.cs]: LastMessageSentDate* ?

File: samples/android/ParseDeepLinkActivity.cs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+	[Activity(Theme="@android:style/Theme.NoDisplay")]
+	public class ParseDeepLinkActivity : Activity
+	{
+		const string TAG = "ParseDeepLinkActivity";
+
+		protected override void OnCreate(Bundle savedInstanceState)
+		{
+			base.OnCreate(savedInstanceState);
+
+			if (DeepLink != null)
```
Comment (hcrowell) [inline samples/android/ParseDeepLinkActivity.cs]: `if (DeepLink == null)` ... if deep link is null, then start the main activity?

**Comment by hcrowell** [comment]
🍊 🍍 🍓  after reviewing comments above ^
nice job!

**Pull request merged**